src/Orleans.Runtime/Hosting/CoreHostingExtensions.cs

```diff
@@ -38,6 +39,24 @@ public static ISiloHostBuilder ConfigureDefaults(this ISiloHostBuilder builder)
             });
         }
 
+        /// <summary>
+        /// Add <see cref="Activity.Current"/> propagation through grain calls.
+        /// Note: according to <see cref="ActivitySource.StartActivity(string, ActivityKind)"/> activity will be created only when any listener for activity exists <see cref="ActivitySource.HasListeners()"/> and <see cref="ActivityListener.Sample"/> returns <see cref="ActivitySamplingResult.PropagationData"/>.
+        /// </summary>
+        /// <param name="builder">The builder.</param>
+        /// <returns>The builder.</returns>
+        public static ISiloBuilder AddActivityPropagation(this ISiloBuilder builder)
+        {
+            if (Activity.DefaultIdFormat != ActivityIdFormat.W3C)
```

Comment:
I am running into systems that have `DefaultIdFormat` set to hierarchical, which cause this to throw.  It appears that https://devblogs.microsoft.com/aspnet/improvements-in-net-core-3-0-for-troubleshooting-and-monitoring-distributed-apps/ and other posts such as https://jimmybogard.com/building-end-to-end-diagnostics-and-tracing-a-primer-trace-context/ are recommending
`Activity.DefaultIdFormat = ActivityIdFormat.W3C;`

Obviously the ClientBuilderExtension would need to change as well.

The systems in question are server2016 with dotnet3.1 runtime.  dotnet5.0 is not installed.

Reply:
Hello, @amccool, we have this check because internal implementation of context uses some w3c specific features, like scope, tags, etc. So, to use this extension you must change default format into W3C.

Reply:
Please reconsider limiting to only W3C. Many systems rely on hierarchical IDs and it's set for entire app. Hierarchical IDs help with logs stored in DBs, since prefixes are matches use index. For example: _startswith "|f728de91a402224da40046d1234e83c5.3b48459462d8e54e."_

Reply:
The default `DistributedContextPropagator` will handle both hierarchical and W3C ActivityId formats.